Python/pylifecycle.c

```diff
@@ -1752,13 +1757,26 @@ finalize_interp_types(PyInterpreterState *interp)
     _PyUnicode_ClearInterned(interp);
 
     _PyDict_Fini(interp);
-    _PyList_Fini(interp);
     _PyTuple_Fini(interp);
 
     _PySlice_Fini(interp);
 
     _PyUnicode_Fini(interp);
     _PyFloat_Fini(interp);
+
+#if defined(Py_GIL_DISABLED)
+    HEAD_LOCK(&_PyRuntime);
+    _PyThreadStateImpl *tstate = (_PyThreadStateImpl *)interp->threads.head;
+    while (tstate != NULL) {
+        finalize_free_lists(&tstate->freelist_state);
+        tstate = (_PyThreadStateImpl *)tstate->base.next;
+    }
+    HEAD_UNLOCK(&_PyRuntime);
+#else
+    // Only free-lists per interpreter are existed.
```

Comment:
How about a comment for the free-threaded build? Also, perhaps try to add some more context for the "GIL comment".

```suggestion
    // Finalize all per-thread free-lists.
    HEAD_LOCK(&_PyRuntime);
    _PyThreadStateImpl *tstate = (_PyThreadStateImpl *)interp->threads.head;
    while (tstate != NULL) {
        finalize_free_lists(&tstate->freelist_state);
        tstate = (_PyThreadStateImpl *)tstate->base.next;
    }
    HEAD_UNLOCK(&_PyRuntime);
#else
    // No per-thread free-lists in GIL builds;
    // we only need to finalize per-interpreter free-lists.
```

Reply:
Thanks it's nice